src/main/java/org/aszjch/demoapp/domain/articlefile/ArticleFileServiceImpl.java

```diff
@@ -0,0 +1,20 @@
+package org.aszjch.demoapp.domain.articlefile;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.aszjch.demoapp.domain.articlefile.port.ArticleFileAssignmentService;
+import org.aszjch.demoapp.domain.articlefile.port.ArticleFileStorageService;
+
+@Slf4j
+@RequiredArgsConstructor
+public class ArticleFileServiceImpl implements ArticleFileService {
+
+    private final ArticleFileStorageService articleFileStorageService;
+    private final ArticleFileAssignmentService articleFileAssignmentService;
+
+    @Override
+    public void upload(final ArticleFile articleFile) {
+        articleFileStorageService.store(articleFile);
+        articleFileAssignmentService.assignFile(articleFile);
```

Comment:
Jesteś pewien że coś co uploaduje powinno przypisywać plik? a nie powinien to robić wywołujący ten serwis u siebie?

Reply:
To jest bardziej klasa sterująca tak naprawdę. Jej by się przydało lepszą nazwę znaleźć. Próbowałem coś się do niej dobrać, ale zakurwiłem głową w circular dependency i uznałem, że zacznę od poprawienia reszty. Jutro pomyślę co da się z tym zrobić